torchvision/models/detection/roi_heads.py

```diff
@@ -479,6 +479,50 @@ def paste_masks_in_image(masks, boxes, img_shape, padding=1):
     return ret
 
 
+def check_target_item(target, key, dtype, shape, shape_string=None):
+    """
+    Checks that a key in target corresponds to a Tensor with correct shape and
+    dtype.
+
+    Args:
+        target (Dict[str, Tensor]): target for a training instance containing
+            the Tensors to be checked
+        key (str): key for the Tensor to be checked in target.
+        dtype (Union[torch.dtype, Tuple[torch.dtype]]): the expected dtype of
+            the Tensor. Can be either a type or a tuple of acceptable types.
+        shape (Tuple): the expected shape of the Tensor. This function checks
+            the number of dimensions and any non-None values of this tuple.
+        shape_string (Optional[str]): optional string for exception messages.
+            If messages. If not specified, shape will be cast into string and
+            used instead.
+
+        Raises:
+            ValueError if the Tensor fails a check.
+        """
+    if not isinstance(dtype, tuple):
+        dtype = (dtype,)
```

Comment:
torchscript doesn't support arguments with multiple types, so it might be better to always let `dtype` be a tuple here

Reply:
This is tricky.

I know that passing `int` works for dtype, but I think this might be a historical artifact.

Also, I would recommend using `List[int]` instead of a `Tuple`, as a tuple needs a fixed (known) size.


cc @eellison can we pass something else as `int` for the dtype type annotation in torchscript?

Reply:
Fix in here https://github.com/pytorch/pytorch/pull/39741